libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3409,16 +3413,22 @@ string YulUtilFunctions::arrayConversionFunction(ArrayType const& _from, ArrayTy
 
 	return m_functionCollector.createFunction(functionName, [&]() {
 		Whiskers templ(R"(
-			function <functionName>(value<?fromCalldataDynamic>, length</fromCalldataDynamic>) -> converted {
+			function <functionName>(value<?fromCalldataDynamic>, length</fromCalldataDynamic>) -> converted <?toCalldataDynamic>, outLength</toCalldataDynamic> {
 				<body>
+				<?toCalldataDynamic>
+					outLength := <length>
+				</toCalldataDynamic>
 			}
 		)");
 		templ("functionName", functionName);
 		templ("fromCalldataDynamic", _from.dataStoredIn(DataLocation::CallData) && _from.isDynamicallySized());
+		templ("toCalldataDynamic", _to.dataStoredIn(DataLocation::CallData) && _to.isDynamicallySized());
+		templ("length", _from.isDynamicallySized() ? "length" : _from.length().str());
```

Comment:
``length`` will never be used for ``!_from.isDynamicallySized()``, right? Doesn't really hurt, though.